contracts/contract/deposit/RocketDepositPool.sol

```diff
@@ -150,28 +164,51 @@ contract RocketDepositPool is RocketBase, RocketDepositPoolInterface, RocketVaul
         uint256 balance = rocketVault.balanceOf("rocketDepositPool");
         uint256 totalEther = 0;
         // Calculate minipool assignments
-        uint256 maxAssignments = rocketDAOProtocolSettingsDeposit.getMaximumDepositAssignments();
-        MinipoolAssignment[] memory assignments = new MinipoolAssignment[](maxAssignments);
-        MinipoolDeposit depositType = MinipoolDeposit.None;
+
         uint256 count = 0;
         uint256 minipoolCapacity = 0;
-        for (uint256 i = 0; i < maxAssignments; ++i) {
-            // Optimised for multiple of the same deposit type
-            if (count == 0) {
-                (depositType, count) = rocketMinipoolQueue.getNextDeposit();
-                if (depositType == MinipoolDeposit.None) { break; }
-                minipoolCapacity = rocketDAOProtocolSettingsMinipool.getDepositUserAmount(depositType);
+        uint256 depositValueForAssignments = msg.value;
+        uint256 socializedAssignments = rocketDAOProtocolSettingsDeposit.getMaximumDepositAssignments();
+        MinipoolAssignment[] memory assignments = new MinipoolAssignment[](maxAssignments);
+
+        // Prepare half deposit assignments
+        count = rocketMinipoolQueue.getLength(MinipoolDeposit.Half);
```

Comment:
If we could iterate across the deposit type, we wouldn't need to copy-paste the code for half/full